specs/makerequest.yaml

```diff
@@ -0,0 +1,39 @@
+name:  makerequest
+doc: |-
+  A machine that creates a HTTP request to Open Weather Map API.
+patternsyntax: json
+nodes:
+  start:
+    branching:
+      type: message
+      branches:
+      - pattern: |
+          {"make":"request"}
+        target: getweather
+  getweather:
+    action:
+      interpreter: ecmascript
+      source: |-
+        _.out({"to":"http","request":{"url":"https://samples.openweathermap.org/data/2.5/weather?zip="+String(_.bindings.zipcode)+"&appid="+String(_.bindings.apk)}});
+        return {};
+    branching:
+      branches:
+      - target: listen
+  listen:
+    branching:
+      type: message
+      branches:
+      - pattern: |
+          {"request":{"url":"?url"}, "body": "?body"}
+        target: report
+  report:
+    action:
+      interpreter: ecmascript
+      source: |-
+        var body = JSON.parse(_.bindings["?body"]);
+        delete _.bindings["?body"];
+        _.out({temprature: parseInt(((body.main.temp - 273.15)*9/5+32))});
+        return {};
```

Comment:
Change this to return _.bindings so the initial bindings will not be lost.  Make sure you also delete binding for `?url`.